hadoop-tools/hadoop-azure/src/main/java/org/apache/hadoop/fs/azure/AzureNativeFileSystemStore.java

```diff
@@ -1698,19 +1721,32 @@ private static void storeLinkAttribute(CloudBlobWrapper blob,
         OLD_LINK_BACK_TO_UPLOAD_IN_PROGRESS_METADATA_KEY);
   }
 
-  private static String getLinkAttributeValue(CloudBlobWrapper blob)
+  private String getLinkAttributeValue(CloudBlobWrapper blob)
       throws UnsupportedEncodingException {
-    String encodedLinkTarget = getMetadataAttribute(blob,
+    String encodedLinkTarget = getMetadataAttribute(blob.getMetadata(),
         LINK_BACK_TO_UPLOAD_IN_PROGRESS_METADATA_KEY,
         OLD_LINK_BACK_TO_UPLOAD_IN_PROGRESS_METADATA_KEY);
     return decodeMetadataAttribute(encodedLinkTarget);
   }
 
-  private static boolean retrieveFolderAttribute(CloudBlobWrapper blob) {
+  private boolean retrieveFolderAttribute(CloudBlobWrapper blob) {
     HashMap<String, String> metadata = blob.getMetadata();
-    return null != metadata
-        && (metadata.containsKey(IS_FOLDER_METADATA_KEY) || metadata
-            .containsKey(OLD_IS_FOLDER_METADATA_KEY));
+    if (null != metadata) {
```

Comment:
Can we reuse **getMetadataAttribute()** API here ?

Something like this :

return null!=getMetadataAttribute(metadata, IS_FOLDER_METADATA_KEY, OLD_IS_FOLDER_METADATA_KEY);

Reply:
Reason why that was not done is to avoid extra call Map#get().  retrieveFolderAttribute() will be a very often called API. So I did not want to make any extra calls as part of this change, specially in default case of 'fs.azure.blob.metadata.key.case.sensitive' false.
Also practically it may not possible but still its possible that a Map contains a given Key but with value as null. What we used to check at this API was/is just check whether key is present in Map or not.

Reply:
agree